Solutions/Digital Shadows/Data Connectors/Digital Shadows/DigitalShadowsConnectorAzureFunction/DS_api.py

```diff
@@ -0,0 +1,89 @@
+""" handles all DS api related functions """
+import logging
+import requests
+import base64
+from urllib.parse import urlparse
+
+logger = logging.getLogger("DS_api")
+
+class api:
+
+    def __init__(self, id, key, secret, url):
+        """ 
+            constructer initializes the DS creds and creates passkey.
+            Parses the url recieved from user.
+        """
+        u = urlparse(url)
+
+        self.url = "https://" + u.netloc + u.path + "/"
+        passkey = key + ":" + secret
+        self.id = id
+        self.b64val = base64.b64encode(bytes(passkey, 'utf-8')).decode("ascii")
+        self.session = requests.Session()
+        self.session.headers.update({
+        "Authorization": "Basic %s" % self.b64val,
+        "searchlight-account-id": "%s" % self.id,
+        "User-Agent": "DigitalShadowsAzureSentinelIntegration"
+        })
+
+    def get_alerts(self, alert_ids):
+        """ 
+            function for getting alerts using id
+        """
+
+        alert_url = self.url + "alerts"
+        params = dict(id=alert_ids)
+        response = self.session.get(alert_url, params=params)
+        logger.info("Alerts response code: %s" % response.status_code)
+        return response
```

Comment:
all of these methods should be doing:

```suggestion
        response.raise_for_status()
        return response.json()
```